crates/sputnik/src/session.rs

```diff
@@ -129,10 +138,20 @@ impl Session {
 }
 
 /// returns sha256 digest of the directory the tool was executed from.
-fn get_cwd_hash() -> Result<String, SputnikError> {
-    let current_dir = current_dir()?;
-    let current_dir_string = current_dir.to_string_lossy();
-    let current_dir_bytes = current_dir_string.as_bytes();
+fn get_cwd_hash(current_dir: &Utf8PathBuf) -> String {
+    format!("{:x}", Sha256::digest(current_dir.as_str().as_bytes()))
+}
+
+/// returns sha256 digest of the repository the tool was executed from.
+fn get_repo_hash(current_dir: &Utf8PathBuf) -> Option<String> {
+    let repo = Repository::discover(current_dir);
+    if let Ok(repo) = repo {
+        if let Ok(remote) = repo.find_remote("origin") {
+            if let Some(remote_url) = remote.url() {
+                return Some(format!("{:x}", Sha256::digest(remote_url.as_bytes())));
```

Comment:
I don't think there's any security implications of hashing the remote, right? Mostly thinking of the issues we had in the past of user/passwords being reported from the remote url. I don't think there's any chance of that here, but just mentioning for my own sanity.

Reply:
thagine good news is that sha-256 is a one-way operation, if someone could reverse the hashing function it would be a [preimage attack](https://en.wikipedia.org/wiki/Preimage_attack). if those were possible we'd have some pretty big problems as an industry 😆